internal/log/log.go

```diff
@@ -0,0 +1,62 @@
+package log
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+type logKey struct{}
+
+type Log struct {
+	op    string
+	start time.Time
+
+	level logrus.Level
+	L     *logrus.Entry
+}
+
+func (l *Log) Done() {
```

Comment:
@kevpar - Kinda took both approaches here. Have both the "start" message in `NewContext` and I log the `duration` in the end message. Really all we need to do is see the end message in automated logging. It will have the "error" if present as well as the timestamps of the operation